### PR TITLE
Set thread's group in rb_thread_create

### PR DESCRIPTION
**Pull request by ahmadsherif**

### CODE REVIEW

File: spec/ruby/optional/capi/thread_spec.rb [hunk 1]
```diff
@@ -84,5 +84,9 @@ def call_capi_rb_thread_wakeup
 
       lambda { thr.join }.should raise_error(NotImplementedError)
     end
+
+    it "sets the thread's group" do
+      @t.rb_thread_create(lambda {}, nil).group.should_not be_nil
```
Comment (YorickPeterse) [inline spec/ruby/optional/capi/thread_spec.rb]: Is it possible here to test for the type that it _does_ return, instead of merely testing if it's not nil?

Comment (ahmadsherif) [inline spec/ruby/optional/capi/thread_spec.rb]: Sure, done.

File: spec/ruby/optional/capi/thread_spec.rb [hunk 2]
```diff
@@ -84,5 +84,10 @@ def call_capi_rb_thread_wakeup
 
       lambda { thr.join }.should raise_error(NotImplementedError)
     end
+
+    it "sets the thread's group" do
+      thread_group = @t.rb_thread_create(lambda {}, nil).group
+      thread_group.class.should == ThreadGroup
```
Comment (YorickPeterse) [inline spec/ruby/optional/capi/thread_spec.rb]: Being picky here, but you can use `.....group.should be_an_instance_of(ThreadGroup)` if I'm not mistaken.

Comment (ahmadsherif) [inline spec/ruby/optional/capi/thread_spec.rb]: No problem, done.

**Comment by YorickPeterse** [comment]
Thanks!

**Pull request merged**